src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/StandardClassLibraryFormatter.cs

```diff
@@ -596,4 +596,38 @@ public Lazy<T> Deserialize(ref MessagePackReader reader, MessagePackSerializerOp
             }
         }
     }
+
+    /// <summary>
+    /// Serializes any instance of <see cref="Type"/> by its <see cref="Type.AssemblyQualifiedName"/> value.
+    /// </summary>
+    /// <typeparam name="T">The <see cref="Type"/> class itself or a derived type.</typeparam>
+    public sealed class TypeFormatter<T> : IMessagePackFormatter<T>
+        where T : Type
+    {
+        public static readonly IMessagePackFormatter<T> Instance = new TypeFormatter<T>();
+
+        private TypeFormatter()
+        {
+        }
+
+        public void Serialize(ref MessagePackWriter writer, T value, MessagePackSerializerOptions options)
+        {
+            if (value is null)
+            {
+                writer.WriteNil();
+            }
+
+            writer.Write(value.AssemblyQualifiedName);
```

Comment:
Missing an `else` here. You don't want to continue writing if `value` is null.

Reply:
Missing a test too, clearly.